File: app/src/main/java/com/app/cheffyuser/cart/fragments/CustomOrderFragment.java
package com.app.cheffyuser.cart.fragments;


import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;

import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.app.cheffyuser.R;
import com.app.cheffyuser.cart.activities.CustomOrderCheckoutActivity;
import com.app.cheffyuser.cart.activities.ItemCartActivity;
import com.app.cheffyuser.cart.adapter.CustomOrderAdapter;

/**
 * A simple {@link Fragment} subclass.
 */
public class CustomOrderFragment extends Fragment {


    public CustomOrderFragment() {
        // Required empty public constructor
    }

    String foodItemList[]={"Grilled salmon","Pasta Ham  ",};
    double foodPriceList[]={96.00,120.00};
    int imgList[]={R.drawable.ic_item_1,R.drawable.ic_item_2};
    RecyclerView recyclerView;




    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view= inflater.inflate(R.layout.fragment_custom_order, container, false);

        LinearLayout itemCart=view.findViewById(R.id.layout_custom_order_cart);

        // get the reference of RecyclerView
        recyclerView = view.findViewById(R.id.recycler_view);


        // set a LinearLayoutManager with default vertical orientation
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(getActivity());

        // set a LinearLayoutManager with default Horizontal orientation
        // LinearLayoutManager linearLayoutManager = new LinearLayoutManager(getApplicationContext(),LinearLayoutManager.HORIZONTAL,false);

        recyclerView.setLayoutManager(linearLayoutManager);

        //  call the constructor of CustomAdapter to send the reference and data to Adapter
        CustomOrderAdapter customAdapter = new CustomOrderAdapter(getActivity(),foodItemList,foodPriceList, imgList);
        recyclerView.setAdapter(customAdapter); // set the Adapter to RecyclerView



        itemCart.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent=new Intent(getActivity(), CustomOrderCheckoutActivity.class);
                startActivity(intent);
            }
        });

        return view;
    }

}

File: app/src/main/java/com/app/cheffyuser/cart/fragments/CustomOrderFragment.kt
package com.app.cheffyuser.cart.fragments


import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.app.cheffyuser.BuildConfig
import com.app.cheffyuser.R
import com.app.cheffyuser.cart.activities.CustomOrderListActivity
import com.app.cheffyuser.cart.adapter.CustomOrderAdapter
import com.app.cheffyuser.home.adapter.RecyclerItemClickListener
import com.app.cheffyuser.home.model.CustomPlateResponseData
import com.app.cheffyuser.home.viewmodel.HomeViewModel
import com.app.cheffyuser.networking.Status
import com.app.cheffyuser.utils.createSnack
import com.app.cheffyuser.utils.hideView
import com.app.cheffyuser.utils.loadAnim
import com.app.cheffyuser.utils.showView
import kotlinx.android.synthetic.main.float_viewcart.*
import kotlinx.android.synthetic.main.fragment_custom_order.*
import kotlinx.android.synthetic.main.item_loading.*
import kotlinx.android.synthetic.main.no_item_layout.*
import timber.log.Timber


/**
 * A simple [Fragment] subclass.
 */
class CustomOrderFragment : Fragment() {

    private lateinit var customOrderAdapter: CustomOrderAdapter

    private val vm: HomeViewModel by lazy {
        ViewModelProviders.of(getActivity()!!).get(HomeViewModel::class.java)
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_custom_order, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)


        no_item_text.text = "Custom orders appear here"
        layout_item_cart.hideView()

        setupCustomOrderList()

        getBasket()

        swipeToRefresh.setOnRefreshListener {
            setupCustomOrderList()
        }
    }

    private fun setupCustomOrderList() {
        val lm = LinearLayoutManager(activity!!, RecyclerView.VERTICAL, false)
        recycler_view.layoutManager = lm
        recycler_view.setHasFixedSize(true)
        recycler_view.animate()

        recycler_view.hideView()
        noitem_layout.hideView()
        loader_layout.showView()

        vm.getCustomPlates().observe(this, Observer {
            swipeToRefresh?.isRefreshing = false
            val data = it.data?.customPlateResponseData

            when (it.status) {
                Status.ERROR -> {
                    recycler_view.hideView()
                    noitem_layout.showView()
                    loader_layout.hideView()

                    if (BuildConfig.DEBUG)
                        createSnack(ctx = activity!!, txt = "Debug only: No custom orders")

                    Timber.d("$it")

                }
                Status.SUCCESS -> {
                    recycler_view.showView()
                    recycler_view.loadAnim()

                    noitem_layout.hideView()
                    loader_layout.hideView()

                    if (!data!!.isNullOrEmpty()) {
                        customOrderAdapter = CustomOrderAdapter(
                            activity!!,
                            data.toMutableList(),
                            object : RecyclerItemClickListener {
                                override fun modelClick(model: Any) {
                                    model as CustomPlateResponseData

                                    val bids = model.customPlateAuction!!.bidCount!!

                                    if (bids > 0) {
                                        val intent = CustomOrderListActivity.newIntent(activity!!)
                                        intent.putExtra(
                                            CustomOrderListActivity.CUSTOM_PLATE_ID,
                                            model.customPlateAuction.id!!
                                        )
                                        activity?.startActivity(intent)

                                    } else {
                                        createSnack(
                                            ctx = activity!!,
                                            txt = "Your order has not received bids yet, we'll notify you",
                                            txtAction = "Refresh",
                                            action = View.OnClickListener {
                                                setupCustomOrderList()
                                            }
                                        )
                                    }
                                }
                            })
                        recycler_view.adapter = customOrderAdapter

                    } else {
                        recycler_view.hideView()
                        noitem_layout.showView()
                        loader_layout.hideView()
                    }
                }
                Status.LOADING -> {
                }
            }
        })
    }


    private fun getBasket() {
        vm.getBasket().observe(this, Observer {
            val data = it.data

            when (it.status) {
                Status.SUCCESS -> {
                    if (!data!!.items.isNullOrEmpty()) {
                        layout_item_cart.showView()
                        tv_total.text = "$" + "${data.total}"
                        tv_count.text = "${data.items?.count()}"

                    } else {
                        layout_item_cart.hideView()
                    }
                }
            }
        })


    }

}
